File: backend/backend/web/api/game/schema/game.py
from backend.web.api.company.schema import CompanyInDB
from backend.web.api.game.schema import GameInDB
from backend.web.api.genre.schema import GenreInDB
from backend.web.api.platform.schema import PlatformInDB
from backend.web.api.sale.schema.sale import Sale
from backend.web.api.user.schema import UserInDB


class Game(GameInDB):
    created_by_company: CompanyInDB
    created_by_user: UserInDB
    genres: list[GenreInDB] = []
    platforms: list[PlatformInDB] = []
    sales: list[Sale] = []

File: frontend/src/client/models/Game.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { CompanyInDB } from './CompanyInDB';
import type { GenreInDB } from './GenreInDB';
import type { PlatformInDB } from './PlatformInDB';
import type { Sale } from './Sale';
import type { UserInDB } from './UserInDB';

export type Game = {
    id: string;
    title: string;
    released_at: string;
    created_at: string;
    created_by_company: CompanyInDB;
    created_by_user?: UserInDB;
    genres?: Array<GenreInDB>;
    platforms?: Array<PlatformInDB>;
    sales?: Array<Sale>;
};

